crates/e2e/src/setup/solver.rs

```diff
@@ -0,0 +1,243 @@
+//! Mock solver for testing purposes. It returns a custom solution.
+
+use {
+    app_data::AppDataHash,
+    axum::Json,
+    ethcontract::{common::abi::ethereum_types::Address, jsonrpc::serde::Serialize},
+    model::{
+        order::{BuyTokenDestination, OrderData, OrderKind, SellTokenSource},
+        signature::EcdsaSigningScheme,
+        DomainSeparator,
+    },
+    solvers_dto::{
+        auction::Auction,
+        solution::{Asset, Kind, Solution, Solutions, Trade},
+    },
+    std::{future::Future, net::SocketAddr, sync::Arc},
+    tokio::{
+        signal::{unix, unix::SignalKind},
+        sync::oneshot,
+    },
+    tracing::Instrument,
+    warp::hyper,
+    web3::signing::SecretKeyRef,
+};
+
+pub struct Mock {
+    solution: Solution,
+}
+
+impl Mock {
+    pub fn new(solution: Solution) -> Self {
+        Self { solution }
+    }
+
+    /// Returns the specified solution with the same order UID as the order UID
+    /// in the auction.
+    pub async fn solve(&self, auction: Auction) -> Solutions {
+        let mut solution = self.solution.clone();
+        // Return the same order UID as the order UID in the auction
+        solution
+            .trades
+            .iter_mut()
+            .filter_map(|trade| match trade {
+                Trade::Fulfillment(fulfillment) => Some(fulfillment),
+                Trade::Jit(_) => None,
+            })
+            .zip(auction.orders.iter())
+            .for_each(|(fulfillment, order)| fulfillment.order = order.uid);
+
+        Solutions {
+            solutions: vec![solution],
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Config {
+    pub addr: SocketAddr,
+    pub solution: Solution,
+}
+
+pub async fn run_mock(config: Config, bind: Option<oneshot::Sender<SocketAddr>>) {
```

Comment:
I think `bind` can be non-optional since there should be no need to spawn a solver of which you don't want to know the port of.